controllers/resources/global.go

```diff
@@ -158,29 +174,33 @@ func (r *Global) Reconcile(ctx context.Context, request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	shouldUpdateStatus := !sets.NewString(tntResource.Status.SelectedTenants...).Equal(tntSet)
-
-	if r.processor.HandlePruning(ctx, tntResource.Status.ProcessedItems, processedItems) {
+	if r.processor.HandlePruning(ctx, tntResource.Status.ProcessedItems.AsSet(), processedItems) {
 		tntResource.Status.ProcessedItems = make([]capsulev1beta2.ObjectReferenceStatus, 0, len(processedItems))
 
 		for _, item := range processedItems.List() {
 			if or := (capsulev1beta2.ObjectReferenceStatus{}); or.ParseFromString(item) == nil {
 				tntResource.Status.ProcessedItems = append(tntResource.Status.ProcessedItems, or)
 			}
 		}
-
-		shouldUpdateStatus = true
 	}
 
-	if shouldUpdateStatus {
-		tntResource.Status.SelectedTenants = tntSet.List()
+	tntResource.Status.SelectedTenants = tntSet.List()
 
-		if updateErr := r.client.Status().Update(ctx, &tntResource); updateErr != nil {
-			log.Error(updateErr, "unable to update TenantResource status")
-		}
+	log.Info("processing completed")
+
+	return reconcile.Result{Requeue: true, RequeueAfter: tntResource.Spec.ResyncPeriod.Duration}, nil
+}
+
+func (r *Global) reconcileDelete(ctx context.Context, tntResource *capsulev1beta2.GlobalTenantResource) (reconcile.Result, error) {
+	log := ctrllog.FromContext(ctx)
+
+	if tntResource.Spec.PruningOnDelete {
+		r.processor.HandlePruning(ctx, tntResource.Status.ProcessedItems.AsSet(), nil)
+
+		controllerutil.RemoveFinalizer(tntResource, finalizer)
 	}
 
 	log.Info("processing completed")
 
-	return reconcile.Result{Requeue: true, RequeueAfter: tntResource.Spec.ResyncPeriod.Duration}, nil
+	return reconcile.Result{}, nil
```

Comment:
We have to enqueue back always with the desired delay to honour the `resyncAfter` input.

Reply:
@prometherion even in case object being deleted?

Reply:
I'd say yes, the workqueue is taking care of it, and if the resource has been deleted, we got the `apierrors.IsNotFound` utility that is discaring it.

Reply:
done